### ISSUE TITLE
Travis Switched to Ubuntu 14.04 And Everything Is Broken 🔥🔥🔥🔥🔥

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#1021]
```
$ sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install gcc-multilib gcc-4.6-arm-linux-gnueabihf lib32stdc++6 ash bash dash ksh mksh zsh pandoc gdb
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package gcc-4.6-arm-linux-gnueabihf
E: Couldn't find any package by glob 'gcc-4.6-arm-linux-gnueabihf'
E: Couldn't find any package by regex 'gcc-4.6-arm-linux-gnueabihf'
```

I'm not sure why this is failing now, since these packages are in the Travis CI whitelist.

https://github.com/travis-ci/apt-package-whitelist/blob/9439e99bbd4ed813d3ed0e4dfb84bbf1bfb40d95/ubuntu-precise#L1177

**Comment by zachriggle** [comment]
Apparently they switched us to Trusty builds? o_0

```
Get:38 http://archive.ubuntu.com/ubuntu trusty-updates/universe i386 Packages [551 kB]
Get:39 http://archive.ubuntu.com/ubuntu trusty-security/main i386 Packages [762 kB]
Get:41 http://archive.ubuntu.com/ubuntu trusty-security/restricted i386 Packages [17.6 kB]
Get:42 http://archive.ubuntu.com/ubuntu trusty-security/multiverse i386 Packages [4,183 B]
Get:43 http://archive.ubuntu.com/ubuntu trusty-security/universe i386 Packages [237 kB]
Get:44 http://archive.ubuntu.com/ubuntu trusty/main i386 Packages [1,739 kB]
Get:45 http://archive.ubuntu.com/ubuntu trusty/restricted i386 Packages [16.4 kB]
Get:46 http://archive.ubuntu.com/ubuntu trusty/multiverse i386 Packages [172 kB]
Get:47 http://archive.ubuntu.com/ubuntu trusty/universe i386 Packages [7,597 kB]
```

**Comment by zachriggle** [comment]
https://blog.travis-ci.com/2017-08-31-trusty-as-default-status

```
We are planning to begin this process on Tuesday, September 5th, 2017.
```

**Comment by zachriggle** [comment]
LOL.

```
The following packages have unmet dependencies:
 gcc-arm-linux-gnueabihf : Depends: gcc-4.8-multilib-arm-linux-gnueabihf (>= 4.8.2-1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by zachriggle** [comment]
Note that 12.04 on Travis will *end support* on April 1, 2018

**Comment by zachriggle** [comment]
Back on trying this, one of the issues is that the we don't get our own network namespace.

This means that trying to SSH into localhost gives us the `sshd` which is running outside of our container -- which does not read "our" `authorized_hosts` file inside the container.

The solution to this will currently be to use `sudo: required`.

**Comment by zachriggle** [comment]
Fixed via #1155 with the caveat that we switched to "sudo: true", which greatly slows down the tests.

I'd rather have them work and be slow, than be broken.
